articles/152_roslaunch_multi_machine_launch.md

```diff
@@ -0,0 +1,310 @@
+---
+layout: default
+title: ROS 2 Multi-Machine Launching
+permalink: articles/roslaunch_mml.html
+abstract:
+  Robotic systems are often distributed across multiple networked machines.
+  This document describes proposed modifications and enhancements to ROS2's
+  launch system to facilitate launching, monitoring, and shutting
+  down systems spread across multiple machines.
+author: '[Matt Lanting](https://github.com/mlanting)'
+published: false
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Authors: {{ page.author }}
+
+## Purpose
+
+Allow a system of ROS nodes to be launched on a hardware architecture that is
+spread across multiple networked computers and facilitate introspection and
+management of the system from a single machine.
+
+## Justification
+
+Nodes can need to run on different hosts for a variety of reasons.  Some possible
+use cases:
+
+- A large robot could with hosts located physically near the hardware they are
```

Comment:
@mlanting hmm, not sure what you mean here.

Reply:
For example, on a vehicle that has many different types of sensors (cameras, lidars, radars, etc.), you may want to have a separate computer for processing the data from each one of those sets of sensors, and in order to simplify cable routing and network design, you may want to have those computers physically positioned next to them.

Alternately, depending on the chassis of your vehicle, you could have limitations on where it's actually possible to place computers, and so they could be scattered around by necessity.

Reply:
```suggestion
- A large robot could require hosts located physically near the hardware they are
```
@wjwwood Is this sufficient to make the explanation clearer? We could go further by describing the physical layout more, possibly talk about doing edge processing for sensors when you have a robot that's more than a few meters in overall size.

Reply:
If you guys are going for a specifically multi-robot/multi-agent approach, we should probably adopt language that suits the field.

```suggestion
- A large robot could with hosts located physically near the hardware they are
+ - A large robot could request physically local static or dynamic agents to control their hardware, such as cameras or other sensors
```